### PR TITLE
Add cortex m7 register definitions

### PR DESCRIPTION
**Pull request by devBoi76**
This PR adds some basic cortex-m7 register definitions.

I verified that ITM, TPIU, and DBG work on an STM32f746ng-disco board with this sample code to write debug info through SWD:

```zig
const std = @import("std");
const microzig = @import("microzig");
const stm32 = microzig.hal;
const cpu = microzig.cpu;

pub fn initSWO() void {
    // Enable trace in core debug:
    cpu.dbg.DEMCR.modify(.{ .TRCENA = 1 }); // Enable debugging & monitoring

    // Enable ITM:
    cpu.itm.TCR.modify(.{
        .ITMENA = 1, // Enable ITM
        .SYNCENA = 1, // Enabe sync packets
        .TXENA = 1, // Enable DWT packets
        .TraceBusID = 1, // Unique ID for multi-source trace
    });

    // Enable stimulus port 0:
    cpu.itm.TER[0].modify(.{ .STIMENA = 1 });

    // Configure Serial Wire Output (SWO):
    cpu.tpiu.SPPR.modify(.{ .TXMODE = 0b10 }); // NZ mode
    cpu.tpiu.ACPR.modify(.{ .SWOSCALER = 7 }); // Set prescaler (for 2MHz output @16MHz)

}

// Helper function to write a byte to ITM stimulus port
pub fn writeITM(ch: u8) void {
    // Wait until stimulus port is ready
    while (cpu.itm.STIM[0].read().READ.FIFOREADY == 0) {}

    // Write character
    // cpu.itm.STIM[0].write(.{ .WRITE_U8 = ch });
    const stim: *volatile u8 = @ptrCast(&cpu.itm.STIM[0].raw);
    stim.* = ch;
}

pub fn writeBytes(str: []const u8) void {
    for (str) |ch| {
        writeITM(ch);
    }
}

pub fn main() void {
    initSWO();
    var fmt_buf: [1024]u8 = undefined;
    var i: u32 = 0;
    while (true) : (i += 1) {
        writeBytes(std.fmt.bufPrintZ(&fmt_buf, "({d}) Hello, SWD!\n", .{i}) catch {
            writeBytes("Buf print formatting error!\n");
            continue;
        });

        for (0..10_000) |_| {
            asm volatile ("nop");
        }
    }
}
```

Prints out what you would expect.

### CODE REVIEW

File: core/src/cpus/cortex_m/m7_utils.zig [hunk 1]
```diff
@@ -0,0 +1,59 @@
+const cpu = @import("../cortex_m.zig");
+
+pub fn initSWO(config: struct {
```
Comment (mattnite) [inline core/src/cpus/cortex_m/m7_utils.zig:3]: `init_swo`

File: core/src/cpus/cortex_m/m7_utils.zig [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    // SWOSCALAR = (Asynchronous_Reference_Clock)/(SWO output clock) - 1
+    if (config.swo_output_target_hz > config.tpiu_clock)
+        return error.SwoTargetFreqTooHigh; // would cause int underflow
+
+    cpu.tpiu.ACPR.modify(.{
+        .SWOSCALER = @as(u16, @intCast((config.tpiu_clock / config.swo_output_target_hz) - 1)),
+    });
+}
+
+pub fn writeByteITM(ch: u8, stim_port: u5) void {
```
Comment (mattnite) [inline core/src/cpus/cortex_m/m7_utils.zig:46]: `write_byte_itm`

File: core/src/cpus/cortex_m/m7_utils.zig [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    // Wait until stimulus port is ready
+    while (cpu.itm.STIM[stim_port].read().READ.FIFOREADY == 0) {}
+
+    // Write character
+    const stim: *volatile u8 = @ptrCast(&cpu.itm.STIM[stim_port].raw);
+    stim.* = ch;
+}
+
+pub fn writeBytesITM(str: []const u8, stim_port: u5) void {
```
Comment (mattnite) [inline core/src/cpus/cortex_m/m7_utils.zig:55]: `write_bytes_itm`

**Review by mattnite — CHANGES_REQUESTED** [review]

File: core/src/cpus/cortex_m/m7_utils.zig (hunk 3 above)
Comment (devBoi76) [reply core/src/cpus/cortex_m/m7_utils.zig]: Done

**Review by mattnite — APPROVED** [review]

**Comment by mattnite** [comment]
@devBoi76 please fix the conflicts and then we're ready to merge

**Comment by devBoi76** [comment]
@mattnite Done 👍

**Pull request merged**